bin/inx-chronicle/src/stardust_inx/mod.rs

```diff
@@ -192,3 +193,18 @@ impl HandleEvent<Report<Syncer>> for InxWorker {
         Ok(())
     }
 }
+
+#[async_trait]
+impl HandleEvent<IsHealthy> for InxWorker {
+    async fn handle_event(
+        &mut self,
+        _cx: &mut ActorContext<Self>,
+        IsHealthy(sender): IsHealthy,
+        inx_client: &mut Self::State,
+    ) -> Result<(), Self::Error> {
+        let node_status = NodeStatus::try_from(inx_client.read_node_status(NoParams {}).await?.into_inner())
+            .map_err(InxError::InxTypeConversion)?;
+        sender.send(node_status.is_healthy).ok();
```

Comment:
Additionally we should check the `latest_confirmed_milestone` or whatever it was exactly called (from the node status) against our database (that also ensures that we - at the moment of the request - are still connected to our mongodb instance and that it's responsive).

Reply:
Sounds like a good idea!